nest-tax-backend/src/admin/admin.service.ts

```diff
@@ -175,260 +178,235 @@ export class AdminService {
-            _sum: {
-              amount: true,
-            },
-            _count: {
-              _all: true,
-            },
-            where: {
-              taxId: taxData.id,
-              status: 'SUCCESS',
-            },
-          })
-          const payedFromNoris =
-            typeof norisPayment.uhrazeno === 'number'
-              ? currency(norisPayment.uhrazeno).intValue
-              : currency(norisPayment.uhrazeno.replace(',', '.')).intValue
-          const forPayment =
-            typeof norisPayment.uhrazeno === 'number'
-              ? currency(norisPayment.zbyva_uhradit).intValue
-              : currency(norisPayment.zbyva_uhradit.replace(',', '.')).intValue
-          if (
-            payerData._sum.amount === null ||
-            payerData._sum.amount < payedFromNoris
-          ) {
-            created += 1
-            const createdTaxPayment =
-              await this.prismaService.taxPayment.create({
-                data: {
-                  amount: payedFromNoris - payerData._sum.amount,
-                  source: 'BANK_ACCOUNT',
-                  specificSymbol: norisPayment.specificky_symbol,
-                  taxId: taxData.id,
-                  status: 'SUCCESS',
-                },
-              })
-            const userFromCityAccount =
-              await this.cityAccountSubservice.getUserDataAdmin(
-                taxData.taxPayer.birthNumber,
-              )
-            if (userFromCityAccount) {
-              await this.bloomreachService.trackEventTaxPayment(
-                {
-                  amount: createdTaxPayment.amount,
-                  payment_source: 'BANK_ACCOUNT',
-                  year: taxData.year,
-                },
-                userFromCityAccount.externalId,
-              )
-            }
+    const taxesDataMap = await this.getTaxesDataMap(norisPaymentData)
 
-            // TODO maybe all theese ifs remove, because logic will be in bloomreach
+    // despite the retype, do not trust the data from Noris & approach as if they were all optional
+    await Promise.all(
+      norisPaymentData.map(async (norisPayment) => {
+        try {
+          const taxData = taxesDataMap[norisPayment.variabilny_symbol]
+          if (taxData) {
+            const payerData = await this.prismaService.taxPayment.aggregate({
+              _sum: {
+                amount: true,
+              },
+              _count: {
+                _all: true,
+              },
+              where: {
+                taxId: taxData.id,
+                status: PaymentStatus.SUCCESS,
+              },
+            })
+            const payedFromNoris =
+              typeof norisPayment.uhrazeno === 'number'
+                ? currency(norisPayment.uhrazeno).intValue
+                : currency(norisPayment.uhrazeno.replace(',', '.')).intValue
+            const forPayment =
+              typeof norisPayment.zbyva_uhradit === 'number'
```

Comment:
this seems like a change in logic, is it still just cleanup or does it change behaviour ?